src/psyclone/domain/lfric/lfric_builtins.py

```diff
@@ -408,6 +408,56 @@ def get_scalar_argument_references(self):
                 if arg.is_scalar]
 
 
+@six.add_metaclass(abc.ABCMeta)
+class LFRicXKern(LFRicBuiltIn):
+    '''Abstract class providing functionaliy to convert a field of
+    one type to a field of another type. If [Datatype] (stored in
+    _field_type) is the particular datatype to convert to and
+    [Precision] is the precision of this datatype then the result is
+    `Y = [Datatype](X, [Precision])`. Here `Y` is a field of type
+    [Datatype] and `X` is a field with a different type. The correct
+    [Precision] is picked up from the associated argument.
+
+    '''
+    _field_type = None
+
+    def gen_code(self, parent):
+        '''Generates LFRic API specific PSy code for a call to the
+        [datatype]_X Built-in.
+
+        :param parent: Node in f2pygen tree to which to add call.
+        :type parent: :py:class:`psyclone.f2pygen.BaseGen`
+
+        :raises InternalError: if the _field_type variable is not set \
+            correctly.
+
+        '''
+        # Subclass must set _field_type
+        if not self._field_type:
```

Comment:
All the PRs associated with mixed precision have been old school gen() and f2pygen. I agree we need to move to the new way. I think Iva hasn't yet done field conversion intrinsics so this could/should be picked up when this is done.